x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/conditional/IfConditional.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.predicate.conditional;
+
+import org.elasticsearch.xpack.sql.expression.Expression;
+import org.elasticsearch.xpack.sql.expression.Nullability;
+import org.elasticsearch.xpack.sql.tree.NodeInfo;
+import org.elasticsearch.xpack.sql.tree.Source;
+import org.elasticsearch.xpack.sql.type.DataType;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Helper function (cannot be directly from a query) to model a {@code WHEN <condition> ELSE <result>}
+ * clause of {@link Case} expression
+ */
+public class IfConditional extends Expression {
+
+    private final Expression condition;
+    private final Expression result;
+
+    public IfConditional(Source source, Expression condition, Expression result) {
+        this(source, Arrays.asList(condition, result));
+    }
+
+    private IfConditional(Source source, List<Expression> children) {
```

Comment:
Why is this constructor needed?

Reply:
No reason after all, it's a leftover from a different design. changing it.